### DISCUSSION TITLE
The Notebook Does Not Appear to Be Valid JSON

### DISCUSSION THREAD

**Discussion by jhparkland** [microsoft/vscode-jupyter#9350] [Questions and Answers] [answered]
![image](https://user-images.githubusercontent.com/86312443/158168533-43170d04-699f-4a75-a07d-2af9926606ca.png)

I wrote a notebook in vcode and uploaded it to git hub repo.

However, the notebook file outputs the following errors.

Error: The Notebook Does Not Appear to Be Valid JSON

My English is poor because I am a Korean student.

The context can be awkward because it is through a translator. I'm sorry.

**Comment by amunger** [comment]
Did you change the file at all before uploading it? If you download that file from Github, does it load in VS Code without errors?
If you click "Raw", you can see the json in plain text and try to check for syntax errors in the json.

**Reply by jhparkland** [reply]
If you download that file from Github, does it load in VS Code without errors? 
>> yes 

If you click "Raw", you can see the json in plain text and try to check for syntax errors in the json.
>> Nothing is displayed after moving to a new window.

**Comment by greazer** [comment]
Hi @pwjdgus, from your screenshot it appears that the notebook is not being displayed in GitHub. You did mention that you created the notebook in VS Code. So, does that same notebook that exists on your local machine still load and operate within VS Code? In other words, is it just GitHub that's having trouble showing the contents?

**Comment by jhparkland** [comment]
@greazer Yes, the same laptop can still work in VS Code. Only GitHub can't see the contents well.

**Reply by 2023396RachelMaeLademora** [reply]
hello i have the same problem i would like to ask if how did you solve this problem? i really need help since i need to submit my assignment on friday.

**Reply by rchiodo** [reply]
Double check the raw contents on github. The notebook may have syntax errors in the json or it might even be empty.

**Comment by rchiodo** [comment, marked as answer]
It looks to me like the file on github is empty. Perhaps it didn't upload.

**Comment by AmritRAjAlok** [comment]
i have same issue please help me out

**Reply by jhparkland** [reply]
In my experience, it was a case where the notebook file was damaged. 
Maybe there was a damage during the download process.

**Comment by rushikesh-pinge** [comment]
Okay so what is the solution to this? How to fix it?

**Reply by jhparkland** [reply]
In my case, I pushed the original Jupyter notebook again. I guess the laptop files were probably damaged for some reason.

However, the mystery is that after repeating the process of pushing the laptop file again a few times, it was recognized normally.

**Comment by Latif-Arib** [comment]
I got the same issue, it is because the notebook which is created in vs code will not have proper JSON format. So I read the file and write it again as notebook format through nbformat package.


```
import nbformat

with open('notebook_name.ipynb', 'r', encoding='utf-8') as f:
    notebook = nbformat.read(f, as_version=4)

with open('notebook_name.ipynb', 'w', encoding='utf-8') as f:
    nbformat.write(notebook, f)
```

**Comment by aranyaadheu** [comment]
![image](https://github.com/user-attachments/assets/2d1788af-c725-47cf-aa35-e2f53d645c82)

I got a perfect working solution, just Clear all Outputs and save it. Now you can git push to GitHub easily. It will be working perfectly fine.
